cartoframes/dataset.py

```diff
@@ -40,55 +40,47 @@ class Dataset(object):
 
     DEFAULT_RETRY_TIMES = 3
 
-    def __init__(self, context=None, table_name=None, schema='public', query=None, df=None, geodf=None):
+    def __init__(self, type, data, context=None, schema='public'):
+        self.type = type
+        self.data = data
```

Comment:
We can not use this way! For example:
- With a dataset created by query, the user wants to play with the data. We need at least `query` and `df`
- The user wants to upload a dataset created by a local DataFrame: we need at least `df` and `table name`

Reply:
OK, so we need to talk then :)

The main point is to define the final API use cases. The combinations between query, df, table_name, gdf, etc. Now we are using the `from_x` methods to create the instances, that is a pretty explicit way, but creating the Dataset based on the params passed is not that explicit.

We can have a meeting on Thursday to define and implement the Dataset API :muscle:

Reply:
:+1: 

Sorry for the duplicated thread